frontend/amundsen_application/static/js/components/Table/index.tsx

```diff
@@ -130,61 +167,227 @@ const ShimmeringBody: React.FC<ShimmeringBodyProps> = ({
-  onCollapse?: (rowValues: any, index: number) => void;
+  onExpand?: (rowValues: any, key: string) => void;
+  onCollapse?: (rowValues: any, key: string) => void;
   isSelectedRow: boolean;
+  nestedLevel: number;
 };
-const ExpandingCell: React.FC<ExpandingCellProps> = ({
-  index,
+const ExpandingButton: React.FC<ExpandingButtonProps> = ({
+  rowKey,
   onClick,
   onExpand,
   onCollapse,
   rowValues,
   expandedRows,
   isSelectedRow,
-}: ExpandingCellProps) => {
-  const isExpanded = expandedRows.includes(index);
-  const cellStyling = { width: EXPANDING_CELL_WIDTH };
+  nestedLevel,
+}: ExpandingButtonProps) => {
+  const isExpanded = expandedRows.includes(rowKey);
+  const buttonContainerStyle = {
+    width: `${getExpandingButtonWidth(nestedLevel)}px`,
+  };
 
   return (
-    <td
-      className="ams-table-cell ams-table-expanding-cell"
-      key={`expandingIndex:${index}`}
-      style={cellStyling}
+    <span
+      className="ams-table-expanding-button-container"
+      style={buttonContainerStyle}
     >
       <button
+        key={rowKey}
         type="button"
         className={`btn ams-table-expanding-button ${
-          isSelectedRow && 'is-selected-row'
-        }`}
+          rowValues.isNestedColumn && 'is-nested-column-row'
+        } ${isSelectedRow && 'is-selected-row'}`}
         onClick={() => {
           const newExpandedRows = isExpanded
-            ? expandedRows.filter((i) => i !== index)
-            : [...expandedRows, index];
+            ? expandedRows.filter((k) => k !== rowKey)
+            : [...expandedRows, rowKey];
 
           onClick(newExpandedRows);
 
           if (!isExpanded && onExpand) {
-            onExpand(rowValues, index);
+            onExpand(rowValues, rowKey);
           }
           if (isExpanded && onCollapse) {
-            onCollapse(rowValues, index);
+            onCollapse(rowValues, rowKey);
           }
         }}
       >
         <span className="sr-only">{EXPAND_ROW_TEXT}</span>
         {isExpanded ? <UpIcon /> : <DownIcon />}
       </button>
-    </td>
+    </span>
   );
 };
 
-type RowIndex = number;
+const TableRow: React.FC<TableRowProps> = ({
+  columnKey,
+  currentSelectedKey,
+  columns,
+  rowValues,
+  rowStyles,
+  onExpand,
+  onCollapse,
+  expandRowRef,
+  expandedRows,
+  setExpandedRows,
+  nestedLevel,
+}: TableRowProps) => {
+  const fields = columns.map(({ field }) => field);
+  const expandingButton = (
+    <ExpandingButton
+      rowKey={columnKey}
+      expandedRows={expandedRows}
+      onExpand={onExpand}
+      onCollapse={onCollapse}
+      rowValues={rowValues}
+      onClick={setExpandedRows}
+      isSelectedRow={currentSelectedKey === columnKey}
+      nestedLevel={nestedLevel}
+    />
+  );
+
+  return (
+    <React.Fragment key={columnKey}>
+      <tr
+        className={`ams-table-row ${
+          rowValues.isNestedColumn && 'is-nested-column-row'
+        } ${currentSelectedKey === columnKey && 'is-selected-row'}`}
```

Comment:
Or was this one? Let's do it here too:
```suggestion
        } ${currentSelectedKey === columnKey ? 'is-selected-row' : ''}`}
```